jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/jdbc/JdbcMetadataReader.java

```diff
@@ -236,7 +236,10 @@ private static String findSchema(String schema, DatabaseMetaData dbMeta, SqlDial
 
     private static List<TableMetadata> findTables(String catalog, String schema, List<String> tableFilter, DatabaseMetaData dbMeta, SqlDialect dialect) throws SQLException {
         List<TableMetadata> tables = new ArrayList<>();
-        ResultSet resTables = dbMeta.getTables(catalog, schema, null, null);
+        
+        String[] supportedTableTypes = {"TABLE", "VIEW", "SYSTEM TABLE"};
```

Comment:
@andrehacker  I added the supportedTableTypes because I have seen in postgres that the getTables functions also retrieved the primary keys as additional objects -> leading to expections. It's true that there is some risk involved introducing that filter, but I think the benefits are greater.